slack/src/tools.js

```diff
@@ -11,43 +12,52 @@ export async function userContext(webClient) {
 }
 
 export async function listChannels(webClient) {
-    const publicChannels = await webClient.conversations.list({limit: 100, types: 'public_channel'})
-    const privateChannels = await webClient.conversations.list({limit: 100, types: 'private_channel'})
+    const result = await webClient.conversations.list({limit: 100, types: 'public_channel,private_channel'})
+    const channels = result.channels
+
+    if (!channels || channels.length === 0) {
+        console.log('No channels found')
+        return
+    }
+
+    console.log(`Found ${channels.length} channels`)
 
     try {
         const gptscriptClient = new GPTScript()
-        const elements = [...publicChannels.channels, ...privateChannels.channels].map(channel => {
+        const elements = channels.map(channel => {
             return {
                 name: `${channel.name}`,
-                description: `${channel.purpose.value || ''}`,
+                description: `${channel.name} (ID: ${channel.id})`,
                 contents: channelToString(channel)
             }
         })
-
         const datasetID = await gptscriptClient.addDatasetElements(elements, {
-            name: "slack_channels"
+            name: 'slack_channels',
         })
-        console.log(`Created dataset with ID ${datasetID} with ${publicChannels.channels.length + privateChannels.channels.length} channels`)
+        console.log(`Created dataset with ID ${datasetID} with ${channels.length} channels`)
     } catch (e) {
         console.log('Failed to create dataset:', e)
     }
 }
 
 export async function searchChannels(webClient, query) {
-    const publicResult = await webClient.conversations.list({limit: 100, types: 'public_channel'})
-    const privateResult = await webClient.conversations.list({limit: 100, types: 'private_channel'})
-
-    const publicChannels = publicResult.channels.filter(channel => channel.name.includes(query))
-    const privateChannels = privateResult.channels.filter(channel => channel.name.includes(query))
+    const result = await webClient.conversations.list({limit: 100, types: 'public_channel,private_channel'})
+    const channels = (new Fuse(
```

Comment:
we can improve the results for `searchUsers` as with this as well. maybe its worth doing it in this PR as well?

Reply:
sure I think that makes sense to do here

Reply:
done